packages/core/src/lib/base_protocol.ts

```diff
@@ -93,8 +83,19 @@ export class BaseProtocol implements IBaseProtocol {
       this.multicodec
     ]);
 
-    // Filter the peers based on the specified criteria
-    return filterPeers(allPeersForProtocol, numPeers, maxBootstrapPeers);
+    // Filter the peers based on discovery & number of peers requested
+    const filteredPeers = await filterPeersByDiscovery(
+      allPeersForProtocol,
+      numPeers,
+      maxBootstrapPeers
+    );
+
+    const sortedFilteredPeers = await sortPeersByLatency(
```

Comment:
Documentation of the function needs to be updated.

Reply:
Can you elaborate?

Doc for `sortPeersByLatency` was updated.
For the parent function `getPeers`, it has a generic definition that wasn't falsy. Updated it for more verbosity: 02aeab804782da1306627f73b14b180a5da80ed3

Reply:
@fryorcraken merging this now
please let me know if you would prefer any follow up -- thanks!